assets/src/dashboard/app/views/editorSettings/googleAnalytics/index.js

```diff
@@ -52,28 +52,30 @@ export const TEXT = {
   INPUT_ERROR: __('Invalid ID format', 'web-stories'),
 };
 
-function GoogleAnalyticsSettings({
-  googleAnalyticsId = '',
-  handleUpdateSettings,
-}) {
-  const [analyticsId, setAnalyticsId] = useState(googleAnalyticsId);
+function GoogleAnalyticsSettings({ googleAnalyticsId, handleUpdateSettings }) {
+  const [analyticsId, _setAnalyticsId] = useState('');
   const [inputError, setInputError] = useState('');
 
+  const setAnalyticsId = useCallback((newId) => _setAnalyticsId(newId), []);
+
+  useEffect(() => {
+    setAnalyticsId(googleAnalyticsId);
+  }, [googleAnalyticsId, setAnalyticsId]);
+
   const handleCancelUpdateId = useCallback(() => {
     setAnalyticsId(googleAnalyticsId);
-  }, [googleAnalyticsId]);
+  }, [googleAnalyticsId, setAnalyticsId]);
```

Comment:
Realized that the googleAnalyticsId is coming through subsequently and never seeing the light of day through here and into the input. Setting up useEffect to make sure proper values are passed.